mongodbatlas/fw_provider.go

```diff
@@ -0,0 +1,395 @@
+	if mongodbgovCloud := data.IsMongodbGovCloud.ValueBool(); mongodbgovCloud {
+		data.BaseURL = types.StringValue(MongodbGovCloudURL)
+	}
+	if data.BaseURL.ValueString() == "" {
+		data.BaseURL = types.StringValue(MultiEnvDefaultFunc([]string{
+			"MONGODB_ATLAS_BASE_URL",
+			"MCLI_OPS_MANAGER_URL",
+		}, "").(string))
+	}
+
+	if data.PublicKey.ValueString() == "" {
+		data.PublicKey = types.StringValue(MultiEnvDefaultFunc([]string{
+			"MONGODB_ATLAS_PUBLIC_KEY",
+			"MCLI_PUBLIC_API_KEY",
+		}, "").(string))
+		if data.PublicKey.ValueString() == "" {
+			resp.Diagnostics.AddError(ProviderConfigError, fmt.Sprintf(AttrNotSetError, "public_key"))
+		}
+	}
+
+	if data.PrivateKey.ValueString() == "" {
+		data.PrivateKey = types.StringValue(MultiEnvDefaultFunc([]string{
+			"MONGODB_ATLAS_PRIVATE_KEY",
+			"MCLI_PRIVATE_API_KEY",
+		}, "").(string))
+		if data.PrivateKey.ValueString() == "" {
+			resp.Diagnostics.AddError(ProviderConfigError, fmt.Sprintf(AttrNotSetError, "private_key"))
+		}
+	}
+
+	if data.RealmBaseURL.ValueString() == "" {
+		data.RealmBaseURL = types.StringValue(MultiEnvDefaultFunc([]string{
+			"MONGODB_REALM_BASE_URL",
+		}, "").(string))
+	}
+
+	if data.Region.ValueString() == "" {
+		data.Region = types.StringValue(MultiEnvDefaultFunc([]string{
+			"AWS_REGION",
+			"TF_VAR_AWS_REGION",
+		}, "").(string))
+	}
+
+	if data.StsEndpoint.ValueString() == "" {
+		data.StsEndpoint = types.StringValue(MultiEnvDefaultFunc([]string{
+			"STS_ENDPOINT",
+			"TF_VAR_STS_ENDPOINT",
+		}, "").(string))
+	}
+
+	if data.AwsAccessKeyID.ValueString() == "" {
+		data.AwsAccessKeyID = types.StringValue(MultiEnvDefaultFunc([]string{
+			"AWS_ACCESS_KEY_ID",
+			"TF_VAR_AWS_ACCESS_KEY_ID",
+		}, "").(string))
+	}
+
+	if data.AwsSecretAccessKeyID.ValueString() == "" {
+		data.AwsSecretAccessKeyID = types.StringValue(MultiEnvDefaultFunc([]string{
+			"AWS_SECRET_ACCESS_KEY",
+			"TF_VAR_AWS_SECRET_ACCESS_KEY",
+		}, "").(string))
+	}
+
+	if data.AwsSessionToken.ValueString() == "" {
+		data.AwsSessionToken = types.StringValue(MultiEnvDefaultFunc([]string{
+			"AWS_SESSION_TOKEN",
+			"TF_VAR_AWS_SESSION_TOKEN",
+		}, "").(string))
+	}
+}
+
+func (p *MongodbtlasProvider) DataSources(context.Context) []func() datasource.DataSource {
+	return []func() datasource.DataSource{
+		// NewExampleDataSource,
+	}
+}
+
+func (p *MongodbtlasProvider) Resources(context.Context) []func() resource.Resource {
+	return []func() resource.Resource{
+		// NewExampleResource,
+	}
+}
+
+func NewFrameworkProvider() provider.Provider {
+	return &MongodbtlasProvider{}
+}
+
+func MuxProviderFactory() func() tfprotov6.ProviderServer {
+	return muxProviderFactoryUsingExistingSdkV2(NewSdkV2Provider())
+}
+
+// muxProviderFactoryUsingExistingSdkV2 creates mux server using existing sdk v2 provider passed as parameter, and new framework provider
+func muxProviderFactoryUsingExistingSdkV2(sdkV2Provider *sdkv2schema.Provider) func() tfprotov6.ProviderServer {
```

Comment:
I'd just rename this to `muxedProviderFactory` or `muxedProviderServer`

Reply:
adjusted to muxedProviderFactory, also added comment with context of why this function is defined.